src/ui/utils/hooks/useDynamic.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from 'react';
+import type { FunctionComponent } from 'react';
+import type { IconName } from '../vite-svgr/types/IProps';
+
+export function useDynamic(iconName: IconName) {
+  const [DynamicComponent, setDynamicComponent] = useState<FunctionComponent<
+    React.SVGProps<SVGSVGElement>
+  > | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<unknown>(null);
+  useEffect(() => {
+    setLoading(true);
+    setError(null);
+    const loadSvg = async () => {
+      const module = (await import(`../../../assets/icons/${iconName}.svg?react`)) as {
+        default: FunctionComponent<React.SVGProps<SVGSVGElement>>;
+      };
+      setDynamicComponent(() => module.default);
+    };
```

Comment:
Move all this code into a separate function. It would be a good idea to create some tests for dynamic loading, especially to handle cases where files are missing or similar situations